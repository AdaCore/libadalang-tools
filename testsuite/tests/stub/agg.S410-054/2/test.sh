gnatstub -q -XT=a_val -P agg.gpr
cat src_a/a.adb
