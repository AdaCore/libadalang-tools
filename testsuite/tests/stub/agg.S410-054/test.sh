echo test number 1
cd 1
gnatstub -q -P agg.gpr
cat src_a1/a1.adb
cd ..

echo test number 2
cd 2
gnatstub -q -XT=a_val -P agg.gpr
cat src_a/a.adb
cd ..

echo test number 3
cd 3
rm -f c.obj d.obj
gnatstub -q -P agg1.gpr
echo "----------------"
echo src_c/c.adb
cat  src_c/c.adb
echo "----------------"
echo src_c/cc.adb
cat  src_c/cc.adb
echo "----------------"
echo src_d/d.adb
cat  src_d/d.adb
echo "----------------"
echo src_d/dd.adb
cat  src_d/dd.adb
gnatmake -q -P agg1.gpr 2>&1 | sort
cd ..
