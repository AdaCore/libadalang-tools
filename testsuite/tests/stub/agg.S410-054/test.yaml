description:
    gnatstub test

driver: shell_script
