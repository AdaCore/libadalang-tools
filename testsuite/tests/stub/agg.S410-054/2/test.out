pragma Ada_2012;
package body a is

   ------------------
   -- Do_Something --
   ------------------

   procedure Do_Something (Param_1, Param_2 : My_Integer_Range_10) is
   begin
      --  Generated stub: replace with real body!
      pragma Compile_Time_Warning (Standard.True,
         "Do_Something unimplemented");
      raise Program_Error with "Unimplemented procedure Do_Something";
   end Do_Something;

   -----------------------
   -- Do_Something_Else --
   -----------------------

   procedure Do_Something_Else (Param_1, Param_2 : My_Integer_Range_10) is
   begin
      --  Generated stub: replace with real body!
      pragma Compile_Time_Warning (Standard.True,
         "Do_Something_Else unimplemented");
      raise Program_Error with "Unimplemented procedure Do_Something_Else";
   end Do_Something_Else;

end a;
